hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/common/AbstractCallAttempt.java

```diff
@@ -16,9 +16,229 @@
 
 package com.hedera.node.app.service.contract.impl.exec.systemcontracts.common;
 
+import static java.util.Objects.requireNonNull;
+
+import com.esaulpaugh.headlong.abi.Function;
+import com.esaulpaugh.headlong.abi.Tuple;
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.node.app.service.contract.impl.exec.gas.SystemContractGasCalculator;
+import com.hedera.node.app.service.contract.impl.exec.scope.HederaNativeOperations;
+import com.hedera.node.app.service.contract.impl.exec.scope.VerificationStrategies;
+import com.hedera.node.app.service.contract.impl.exec.scope.VerificationStrategy;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AddressIdConverter;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
+import com.swirlds.config.api.Configuration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.nio.BufferUnderflowException;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+
 /**
- * Base class for system contract call attempts.
- * @Future for now this abstract class is empty, subsequent PRs will add common call attempt functionality
- * once a HasCallAttempt class has been added.
+ * Base class for HTS and HAS system contract call attempts.
  */
-public class AbstractCallAttempt {}
+public class AbstractCallAttempt {
+    private final byte[] selector;
+    protected Bytes input;
+    protected boolean isRedirect;
```

Comment:
This field is `true` iff `redirectAddress` is non-null, right? Do we need it?

Reply:
make sense.  I'll remove and update to a method.